spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java

```diff
@@ -74,7 +74,10 @@
 	 */
 	private final String[] headersToEmbed;
 
-	private final ProvisioningProvider<C, P> provisioningProvider;
+	/**
+	 * {@link ProvisioningProvider} delegated by the downstream binder implementations.
+	 */
+	protected final ProvisioningProvider<C, P> provisioningProvider;
```

Comment:
I am fine with adding that extra generic argument to avoid casting in implementations in subclasses. Will send another push.